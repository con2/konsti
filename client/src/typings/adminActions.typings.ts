import { Game } from 'typings/game.typings';

export const SUBMIT_UPDATE_HIDDEN = 'SUBMIT_UPDATE_HIDDEN';
export const SUBMIT_GET_SETTINGS = 'SUBMIT_GET_SETTINGS';
export const SUBMIT_SELECT_SIGNUP_TIME = 'SUBMIT_SELECT_SIGNUP_TIME';
export const SUBMIT_SET_TEST_TIME = 'SUBMIT_SET_TEST_TIME';
export const SUBMIT_TOGGLE_APP_OPEN = 'SUBMIT_TOGGLE_APP_OPEN';
export const SUBMIT_RESPONSE_MESSAGE = 'SUBMIT_RESPONSE_MESSAGE';

export interface SubmitUpdateHiddenAsync {
  type: typeof SUBMIT_UPDATE_HIDDEN;
  hiddenGames: readonly Game[];
}

export interface SubmitGetSettingsAsync {
  type: typeof SUBMIT_GET_SETTINGS;
  hiddenGames: readonly Game[];
  signupTime: string;
  appOpen: boolean;
}

export interface SubmitSignupTimeAsync {
  type: typeof SUBMIT_SELECT_SIGNUP_TIME;
  signupTime: string;
}

export interface SubmitSetTestTime {
  type: typeof SUBMIT_SET_TEST_TIME;
  testTime: string;
}

export interface SubmitToggleAppOpenAsync {
  type: typeof SUBMIT_TOGGLE_APP_OPEN;
  appOpen: boolean;
}

export interface SubmitResponseMessageAsync {
  type: typeof SUBMIT_RESPONSE_MESSAGE;
  responseMessage: string;
}

export type AdminActionTypes =
  | SubmitUpdateHiddenAsync
  | SubmitGetSettingsAsync
  | SubmitSignupTimeAsync
  | SubmitSetTestTime
  | SubmitToggleAppOpenAsync
  | SubmitResponseMessageAsync;
