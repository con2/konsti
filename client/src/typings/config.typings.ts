export interface Config {
  appName: string;
  SIGNUP_END_TIME: number;
  SIGNUP_OPEN_TIME: number;
  MESSAGE_DELAY: number;
  CONVENTION_NAME: ConventionName;
  CONVENTION_YEAR: string;
  CONVENTION_START_TIME: string;
  DAY_START_TIME: number;
  apiServerURL: string;
  useTestTime: boolean;
  reduxTrace: boolean;
  loadedSettings: string;
  enableAxe: boolean;
  enableWhyDidYouRender: boolean;
  dataUpdateInterval: number;
  noSignupGames: string[];
  revolvingDoorEnabled: boolean;
  tagFilteringEnabled: boolean;
  simpleDetails: boolean;
}

type ConventionName = 'Ropecon' | 'Tracon Hitpoint';
