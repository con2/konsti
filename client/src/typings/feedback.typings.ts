export interface Feedback {
  feedback: string;
  gameId: string;
}

export interface PostFeedbackResponse {
  message: string;
  status: 'success';
}
