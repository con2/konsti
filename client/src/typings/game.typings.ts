export interface Game {
  gameId: string;
  title: string;
  description: string;
  location: string;
  startTime: string;
  mins: number;
  tags: readonly string[];
  genres: readonly string[];
  styles: readonly string[];
  language: string;
  endTime: string;
  people: string;
  minAttendance: number;
  maxAttendance: number;
  gameSystem: string;
  englishOk: boolean;
  childrenFriendly: boolean;
  ageRestricted: boolean;
  beginnerFriendly: boolean;
  intendedForExperiencedParticipants: boolean;
  shortDescription: string;
  revolvingDoor: boolean;
  popularity: number;
  programType: string;
}

export interface DnDUpdatedPositions {
  availableGames?: readonly Game[];
  selectedGames?: readonly Game[];
}

export interface DnDMove {
  index: number;
  droppableId: string;
}

export interface PostGamesUpdateResponse {
  games: Game[];
  message: string;
  status: 'success';
}

export interface GetGamesResponse {
  games: Game[];
  message: string;
  status: 'success';
}

export interface PostHiddenResponse {
  hiddenGames: Game[];
  message: string;
  status: 'success';
}
