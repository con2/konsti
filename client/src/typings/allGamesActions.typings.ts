import { Game } from 'typings/game.typings';

export const SUBMIT_GET_GAMES = 'SUBMIT_GET_GAMES';

export interface SubmitGetGamesAsync {
  type: typeof SUBMIT_GET_GAMES;
  games: readonly Game[];
}

export type AllGamesActionTypes = SubmitGetGamesAsync;
