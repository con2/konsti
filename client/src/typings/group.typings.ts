import { Signup } from 'typings/user.typings';

export interface GroupMember {
  enteredGames: readonly Signup[];
  groupCode: string;
  serial: string;
  signedGames: readonly Signup[];
  username: string;
}

export interface GroupData {
  groupCode: string;
  leader: boolean;
  ownSerial: string;
  username: string;
  leaveGroup?: boolean;
  closeGroup?: boolean;
}

export interface PostGroupResponse {
  groupCode: string;
  message: string;
  status: 'success';
}

export interface GetGroupResponse {
  message: string;
  results: GroupMember[];
  status: 'success';
}
