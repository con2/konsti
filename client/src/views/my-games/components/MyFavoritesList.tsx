import React, { FC, ReactElement } from 'react';
import { useTranslation } from 'react-i18next';
import _ from 'lodash';
import styled from 'styled-components';
import { getStartTimes } from 'utils/getStartTimes';
import { GamesByStartTimes } from './GamesByStartTimes';
import { Game } from 'typings/game.typings';

export interface Props {
  favoritedGames: readonly Game[];
}

export const MyFavoritesList: FC<Props> = (props: Props): ReactElement => {
  const { favoritedGames } = props;
  const { t } = useTranslation();

  const sortedGames: readonly Game[] = _.sortBy(favoritedGames, [
    (favoritedGame) => favoritedGame.startTime,
    (favoritedGame) => favoritedGame.title.toLowerCase(),
  ]);

  const startTimes = getStartTimes(favoritedGames);

  return (
    <div className='my-favorites-list'>
      <h3>{t('favoritedGames')}</h3>
      <MyFavoritesGames>
        {favoritedGames.length === 0 && <span>{t('noFavoritedGames')}</span>}
        {favoritedGames.length !== 0 && (
          <GamesByStartTimes games={sortedGames} startTimes={startTimes} />
        )}
      </MyFavoritesGames>
    </div>
  );
};

const MyFavoritesGames = styled.div`
  padding-left: 30px;
`;
