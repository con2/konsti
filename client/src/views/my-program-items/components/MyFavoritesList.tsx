import { ReactElement } from "react";
import { useTranslation } from "react-i18next";
import { sortBy } from "lodash-es";
import styled from "styled-components";
import { getStartTimes } from "client/utils/getStartTimes";
import { FavoritesByStartTimes } from "./FavoritesByStartTimes";
import { ProgramItem } from "shared/types/models/programItem";
import { RaisedCard } from "client/components/RaisedCard";
import {
  MyProgramHeader,
  MyProgramSecondaryText,
} from "client/views/my-program-items/components/shared";

interface Props {
  favoriteProgramItems: readonly ProgramItem[];
}

export const MyFavoritesList = ({
  favoriteProgramItems,
}: Props): ReactElement => {
  const { t } = useTranslation();

  const sortedProgramItems: readonly ProgramItem[] = sortBy(
    favoriteProgramItems,
    [
      (favoriteProgramItem) => favoriteProgramItem.startTime,
      (favoriteProgramItem) => favoriteProgramItem.title.toLowerCase(),
    ],
  );

  const startTimes = getStartTimes(favoriteProgramItems);

  return (
    <RaisedCard>
      <MyProgramHeader>{t("favoriteProgramItems")}</MyProgramHeader>
      <div>
        {favoriteProgramItems.length === 0 && (
          <MyProgramSecondaryText>
            {t("noFavoriteProgramItems")}
          </MyProgramSecondaryText>
        )}
        {favoriteProgramItems.length !== 0 && (
          <FavoritesByStartTimes
            programItems={sortedProgramItems}
            startTimes={startTimes}
          />
        )}
      </div>
    </RaisedCard>
  );
};
