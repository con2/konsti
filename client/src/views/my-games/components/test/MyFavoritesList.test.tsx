import React from 'react';
import { shallow } from 'enzyme';
import { MyFavoritesList, Props } from '../MyFavoritesList';
import { Game } from 'typings/game.typings';

const favoritedGames: Game[] = [];

describe('MyFavoritesList', () => {
  it('should render correctly', () => {
    const props: Props = { favoritedGames };
    const component = shallow(<MyFavoritesList {...props} />);
    expect(component).toMatchSnapshot();
  });
});
