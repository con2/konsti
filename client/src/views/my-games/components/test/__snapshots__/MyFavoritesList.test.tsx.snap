// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyFavoritesList should render correctly 1`] = `
<div
  className="my-favorites-list"
>
  <h3>
    favoritedGames
  </h3>
  <MyFavoritesList__MyFavoritesGames>
    <span>
      noFavoritedGames
    </span>
  </MyFavoritesList__MyFavoritesGames>
</div>
`;
